### PR TITLE
2020 update

### PR DESCRIPTION
**Pull request by evansiroky**
- update to latest osmosis library
- fix javax import
- fix error causing NPE when a network failure occurs while updating
- fix error caused by not catching URL parsing error

### CODE REVIEW

File: src/main/java/com/conveyal/osmlib/Updater.java [hunk 1]
```diff
@@ -91,7 +90,9 @@ public Diff fetchState(String timescale, int sequenceNumber) {
                 LOG.debug("Checking replication state for timescale {}", timescale);
             }
             sb.append("state.txt");
-            URL url = new URL(sb.toString());
+            String replicationUrl = sb.toString();
```
Comment (landonreed) [inline src/main/java/com/conveyal/osmlib/Updater.java]: Can we call this `planetReplicationUrl`?

File: src/main/java/com/conveyal/osmlib/VanillaExtract.java [hunk 2]
```diff
@@ -105,11 +105,14 @@ public void service(Request request, Response response) throws Exception {
...
             LOG.info("VEX request: {}", uri);
-            int suffixIndex = uri.lastIndexOf('.');
-            String fileType = uri.substring(suffixIndex);
             OutputStream outStream = response.getOutputStream();
             try {
+                int suffixIndex = uri.lastIndexOf('.');
+                String fileType = uri.substring(suffixIndex);
                 String[] coords = uri.substring(1, suffixIndex).split("[,;]");
+                if (coords.length < 4) {
+                    throw new IllegalArgumentException("Must provide a bounding box with 4 coordinates");
```
Comment (landonreed) [inline src/main/java/com/conveyal/osmlib/VanillaExtract.java:114]: This exception should probably be thrown, but there is a TODO in the code below that talks about future work on an extra `buildings` param. Maybe that should just be removed for now?

File: src/main/java/com/conveyal/osmlib/VanillaExtract.java [hunk 3]
```diff
@@ -135,7 +138,6 @@ public void service(Request request, Response response) throws Exception {
                 response.setContentType("text/plain");
                 response.setStatus(HttpStatus.BAD_REQUEST_400);
                 outStream.write("URI format: /min_lat,min_lon,max_lat,max_lon[.pbf|.vex] (all coords in decimal degrees)\n".getBytes());
-                ex.printStackTrace();
```
Comment (landonreed) [inline src/main/java/com/conveyal/osmlib/VanillaExtract.java:138]: Do we not want to print the exception at all? I think we should maybe have separate catch clauses. One for generic exception that prints the exception and one for the IllegalArgumentException, that just prints the error message with URI above.

**Review by landonreed — CHANGES_REQUESTED** [review]
Looks good. Just one blocking change with respect to the catch clause

**Comment by evansiroky** [comment]
Changes should be addressed

**Review by landonreed — APPROVED** [review]

**Pull request merged**